lib/optimizely/config/datafile_project_config.rb

```diff
@@ -304,17 +371,17 @@ def get_variation_id_from_key(experiment_key, variation_key)
       nil
     end
 
-    def get_whitelisted_variations(experiment_key)
+    def get_whitelisted_variations(experiment_id)
       # Retrieves whitelisted variations for a given experiment Key
       #
       # experiment_key - String Key representing the experiment
       #
       # Returns whitelisted variations for the experiment or nil
 
-      experiment = @experiment_key_map[experiment_key]
+      experiment = @experiment_id_map[experiment_id]
       return experiment['forcedVariations'] if experiment
 
-      @logger.log Logger::ERROR, "Experiment key '#{experiment_key}' is not in datafile."
+      @logger.log Logger::ERROR, "Experiment key '#{experiment_id}' is not in datafile."
```

Comment:
It will be helpful to keep both `experiment_id` and `experiment_key` in the log

Reply:
Cant use experiment key here as we are no longer sending key in `get_whitelisted_variations` instead we are sending the ID. If the given ID is wrong we cannot get an experiment and hence we can not get the key.